### PR TITLE
LPS-163597 Implement UI to delete an FDS Custom View

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-163597 Implement UI to delete an FDS Custom View](https://issues.liferay.com/browse/LPS-163597)
- mockup: [Allow users to delete their custom views
](https://www.figma.com/proto/EqgsL82bX7GJfLdZA8iYFv/LPS-130101-allow-users-customize-and-manage-views-in-frontend-dataset?node-id=5%3A63133&scaling=min-zoom&page-id=5%3A63132)

### What is the goal of this PR?

In this PR, we are:
- Adding a REST endpoint to delete a custom view (pinging @LuismiBarcos and @ccorreagg to review this part)
- Adding UI to delete a custom view (pinging @diegonvs  and @tinycarol to review this part)

After deletion, UI resets to the default view. 

### What does it look like?

after

### Steps to reproduce
1. Deploy `frontend-data-set-sample-web`.
2. Place `Frontend Data Set Sample` widget on any page.

### CODE REVIEW

**Comment by markocikos** [comment]
ci:test:relevant

**Review by tinycarol — APPROVED** [review]
UI code LGTM!

**Review by diegonvs — APPROVED** [review]

**Comment by diegonvs** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java [hunk 1]
```diff
@@ -70,6 +71,45 @@
...
+
+			return Response.ok(
+			).build();
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+
+		return Response.status(
+			Response.Status.INTERNAL_SERVER_ERROR
```
Comment (LuismiBarcos) [inline modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java:109]: Maybe in the future, we could check the possible errors and return a proper description and error. I assume that this endpoint is for internal use so we could change the response and the clients will not suffer a breaking change in the response. 
If not, we should check the errors to return the proper information.

**Review by LuismiBarcos — APPROVED** [review]
I approve the PR because I assume that this is for internal use.
I made some comments with a few possible changes just in case you think it is necessary.

But looks good to me :)

File: modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java:109]: Yes, endpoint is for internal use, so this is not a blocker for this PR. But, it's a good suggestion! I think we can extend error reporting as a separate task, I added the task to [technical improvements checklist](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2002681985/FDS+technical+improvements+checklist#FDS-API).

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**